cybershell/memory.py

```
@@ -0,0 +1,74 @@
+    def _reindex(self):
+        corpus = [f"{i.title} {i.content} {' '.join(i.tags)}" for i in self.items]
+        self._matrix = self.vec.fit_transform(corpus) if corpus else []
+
+    def search(self, query: str, top_k: int = 8):
+        if not self.items: return []
+        qv = self.vec.transform([query])[0]
+        scored = [(it, self.vec.cosine(qv, self._matrix[idx])) for idx,it in enumerate(self.items)]
+        scored.sort(key=lambda x: x[1], reverse=True)
+        return [{'item': it, 'score': sc} for it, sc in scored[:top_k]]
+
+    def reinforce(self, titles: List[str], delta: float = 0.1):
```

Ruff (0.12.2)

Missing return type annotation for public function `reinforce`

Add return type annotation: `None`

(ANN201)